out_newrelic.go

```diff
@@ -64,7 +64,7 @@ func FLBPluginFlushCtx(ctx, data unsafe.Pointer, length C.int, tag *C.char) int
 	// output.FLB_ERROR = unrecoverable error, do not try this again.
 	// output.FLB_RETRY = retry to flush later.
 	if err := nrClient.Send(buffer); err != nil {
-		return output.FLB_ERROR
+		return output.FLB_RETRY
```

Comment:
how do you differentiate recoverable errors from irrecoverable?

Reply:
@tejunior - We don't differentiate.  From a technical standpoint, I'm assuming that if the flush fails, it's a temporary condition that will be resolved.  From an operational standpoint, I'm assuming that keeping the logs and retrying (even if it fails forever) is more desirable than dropping them on the floor due to a transient condition.

Reply:
I got you point... my concern is that this returned error can be from a http error code to an Unmarshalling error. So if we get an error like "bad format", or something unrecoverable, maybe we got an endless loop?

Reply:
In this case, we would get an endless loop in the sense that FluentBit would store the records and attempt to send them forever, but that would not block further processing by any part of FluentBit.  The risk is that logs might pile up and consume memory or disk space up to whichever limits have been set.

My concern about parsing the err string returned by nrClient is its variability and difficulty in determining the best path for the various modes of failure.  For example, how should we handle the various classes of HTTP error?  A 403 implies a bad license key - do we keep those logs or drop them?  What about a 500?

Overall, I can't see a condition where we'd want to drop logs. At some point someone will see that there's a problem, investigate, and then they can make the decision to manually drop the logs or not.

Reply:
Hello @AlecIsaacson, this issue got addressed in this [PR. ](https://github.com/newrelic/newrelic-fluent-bit-output/pull/73). Thank you so much for contributing.